pkg/plugin/linuxutil/linuxutil_linux.go

```diff
@@ -61,6 +62,26 @@ func (lu *linuxUtil) SetupChannel(chan *hubblev1.Event) error {
 
 func (lu *linuxUtil) run(ctx context.Context) error {
 	lu.l.Info("Running linuxutil plugin...")
+
+	ethtoolOpts := &EthtoolOpts{
+		errOrDropKeysOnly: true,
+		addZeroVal:        false,
+		limit:             defaultLimit,
+	}
+
+	ethHandle, err := ethtool.NewEthtool()
+	if err != nil {
+		lu.l.Error("Error while creating ethHandle: %v\n", zap.Error(err))
+		return fmt.Errorf("failed to create ethHandle: %w", err)
+	}
+	defer ethHandle.Close()
+
+	ethReader := NewEthtoolReader(ethtoolOpts, ethHandle)
+	if ethReader == nil {
+		lu.l.Error("Error while creating ethReader")
+		return errors.New("error while creating ethReader")
+	}
```

Comment:
One of the reason for keeping the `ethHandle` inside the loop is to make sure we are not bringing down Retina agent for transient error. Do you think the `ethHandle` error is permanent - as in if it fails once, it cannot succeed on retry? 
If you return error here, `PluginManager` will treat it as a fatal error.

Reply:
Brining this out is probably ok, but I would retry few time (with exponential backoff) before reporting failure.

Reply:
I think your cache shouldn't be tied to the lifecycle of the tool, it should persist for agent's lifecycle.